### PR TITLE
DM-29354: Remove deprecated datastores

### PR DESCRIPTION
**Pull request by timj**
These have not been used for nearly 6 months and are not used in any
supported deployments.

### CODE REVIEW

File: doc/lsst.daf.butler/datastores.rst [hunk 1]
```diff
@@ -9,7 +9,7 @@ Datastore Configuration
 A Butler `Datastore` is configured in the ``datastore`` section of the top-level Butler YAML configuration.
 The only mandatory entry in the datastore configuration is the ``cls`` key.
 This specifies the fully qualified class name of the Python class implementing the datastore.
-The default Butler configurations uses the `~datastores.posixDatastore.PosixDatastore`.
+The default Butler configurations uses the `~datastores.fileDatastore.FileDatastore`.
```
Comment (ktlim) [inline doc/lsst.daf.butler/datastores.rst]: "configurations uses" should be either "configurations use" or "configuration uses".

File: doc/lsst.daf.butler/datastores.rst [hunk 2]
```diff
@@ -28,7 +28,8 @@ The supported datastores are:
 File-Based Datastores
 =====================
 
-The file-based datastores (for example `~datastores.posixDatastore.PosixDatastore` and `~datastores.s3Datastore.S3Datastore`) share configuration since they use formatters to serialize datasets to file artifacts using file name template schemes and also support disassembly of :ref:`composite storage classes <daf_butler_storage_classes>`.
+There is a single file-based datastore (`~datastores.fileDatastore.FileDatastore`) that handles local POSIX file system and remote object stores.
+Data access is entirely mediated by the URI used to specify the datastore root and currently support S3 and WebDAV in addition to local files.
```
Comment (ktlim) [inline doc/lsst.daf.butler/datastores.rst]:
```suggestion
Data access is entirely mediated by the URI used to specify the datastore root and currently supports S3 and WebDAV in addition to local files.
```

File: doc/lsst.daf.butler/datastores.rst [hunk 3]
```diff
@@ -28,7 +28,8 @@ The supported datastores are:
 File-Based Datastores
 =====================
 
-The file-based datastores (for example `~datastores.posixDatastore.PosixDatastore` and `~datastores.s3Datastore.S3Datastore`) share configuration since they use formatters to serialize datasets to file artifacts using file name template schemes and also support disassembly of :ref:`composite storage classes <daf_butler_storage_classes>`.
+There is a single file-based datastore (`~datastores.fileDatastore.FileDatastore`) that handles local POSIX file system and remote object stores.
```
Comment (ktlim) [inline doc/lsst.daf.butler/datastores.rst:31]: The main criteria for this datastore are that every URI corresponds to a single (possibly composite) dataset and, vice versa, that every dataset is labeled by a URI?  (As opposed to, say, a SQL-database-like datastore where multiple queries, even if expressed as URIs, might correspond to the same effective dataset.)

**Review by ktlim — APPROVED** [review]

File: doc/lsst.daf.butler/datastores.rst (hunk 3 above)
Comment (timj) [reply doc/lsst.daf.butler/datastores.rst:31]: I've added an explicit comment about that.

**Pull request merged**